New buildbot failure: AMD64 FreeBSD does not have `netbt/l2cap.h`

```
cc -pthread  -fno-strict-overflow -Wsign-compare -g -Og -Wall  -O2 -pipe  -std=c11 -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wstrict-prototypes -Werror=implicit-function-declaration -fvisibility=hidden  -I./Include/internal -I./Include/internal/mimalloc  -I. -I./Include    -fPIC -c ./Modules/syslogmodule.c -o Modules/syslogmodule.o
--- Modules/socketmodule.o ---
./Modules/socketmodule.c:212:12: fatal error: 'netbt/l2cap.h' file not found
  212 | #  include <netbt/l2cap.h>
      |            ^~~~~~~~~~~~~~~
--- Modules/termios.o ---
```

Link: https://buildbot.python.org/#/builders/1617/builds/295
Refs https://github.com/python/cpython/pull/132735

CC @serhiy-storchaka

--- a/Lib/test/test_socket.py
+++ b/Lib/test/test_socket.py
@@ -2647,14 +2647,13 @@ class BasicBluetoothTest(unittest.TestCase):
             socket.BT_POWER
             socket.BT_CHANNEL_POLICY
             socket.BT_CHANNEL_POLICY_BREDR_ONLY
-            socket.BT_PHY
-            socket.BT_PHY_BR_1M_1SLOT
-            socket.BT_MODE
-            socket.BT_MODE_BASIC
-            socket.BT_VOICE
-            socket.BT_VOICE_TRANSPARENT
-            socket.BT_VOICE_CVSD_16BIT
-            socket.BT_CODEC
+            if hasattr(socket, 'BT_PHY'):
+                socket.BT_PHY_BR_1M_1SLOT
+            if hasattr(socket, 'BT_MODE'):
+                socket.BT_MODE_BASIC
+            if hasattr(socket, 'BT_VOICE'):
+                socket.BT_VOICE_TRANSPARENT
+                socket.BT_VOICE_CVSD_16BIT
             socket.L2CAP_LM
             socket.L2CAP_LM_MASTER
             socket.L2CAP_LM_AUTH

--- a/Modules/socketmodule.c
+++ b/Modules/socketmodule.c
@@ -208,7 +208,7 @@ shutdown(how) -- shut down traffic in one or both directions\n\
 #  include <sys/ioctl.h>
 #endif
 
-#ifdef HAVE_BLUETOOTH_H
+#if defined(HAVE_BLUETOOTH_H) && !defined(__FreeBSD__)
 #  include <netbt/l2cap.h>
 #  include <netbt/rfcomm.h>
 #  include <netbt/hci.h>
@@ -8014,6 +8014,7 @@ socket_exec(PyObject *m)
 #endif
     ADD_INT_MACRO(m, BT_SNDMTU);
     ADD_INT_MACRO(m, BT_RCVMTU);
+#ifdef BT_PHY
     ADD_INT_MACRO(m, BT_PHY);
     ADD_INT_MACRO(m, BT_PHY_BR_1M_1SLOT);
     ADD_INT_MACRO(m, BT_PHY_BR_1M_3SLOT);
@@ -8030,15 +8031,24 @@ socket_exec(PyObject *m)
     ADD_INT_MACRO(m, BT_PHY_LE_2M_RX);
     ADD_INT_MACRO(m, BT_PHY_LE_CODED_TX);
     ADD_INT_MACRO(m, BT_PHY_LE_CODED_RX);
+#endif
+#ifdef BT_MODE
     ADD_INT_MACRO(m, BT_MODE);
     ADD_INT_MACRO(m, BT_MODE_BASIC);
     ADD_INT_MACRO(m, BT_MODE_ERTM);
     ADD_INT_MACRO(m, BT_MODE_STREAMING);
     ADD_INT_MACRO(m, BT_MODE_LE_FLOWCTL);
     ADD_INT_MACRO(m, BT_MODE_EXT_FLOWCTL);
+#endif
+#ifdef BT_PKT_STATUS
     ADD_INT_MACRO(m, BT_PKT_STATUS);
+#endif
+#ifdef BT_ISO_QOS
     ADD_INT_MACRO(m, BT_ISO_QOS);
+#endif
+#ifdef BT_CODEC
     ADD_INT_MACRO(m, BT_CODEC);
+#endif
 #endif /* HAVE_BLUETOOTH_BLUETOOTH_H */
 #endif /* USE_BLUETOOTH */
 

